### ISSUE TITLE
iCubGenova09 left hip roll AMO sensor broken

### ISSUE THREAD

**Issue by S-Dafarra** [robotology/icub-tech-support#1348]
### Device name 🤖

iCubGenova09

### Request/Failure description

The left hip roll AMO sensor is broken

### Detailed context

While preparing the demo at Minister della Salute, @Uboldi80 noticed that the red wire of the left hip roll AMO sensor was detached from the connector. As soon as he touched it, the hip roll snapped and, after that, the encode was reading constant values.

### Additional context

Before we figured out that the AMO was broken, @gsisinna and @Uboldi80 changed also an EMS and two 2FOC boards.

**Comment by S-Dafarra** [comment]
cc @lrapetti @mebbaid

**Comment by Uboldi80** [comment]
i have replaced the AMO board in the left leg.
the old one shows evident signs of a short circuit
tomorrow we will align the firmware and carry out the calibration of the coupling.

**Comment by gsisinna** [comment]
Today with @Uboldi80 and @davidetome we tried to calibrate the new AMO board. The first step of analogic calibration went well, but we found some problems with the Nonius Calibration (see [procedures](https://icub-tech-iit.github.io/procedures/tp-amo-calibration/)). We got this result:
![AMO](https://user-images.githubusercontent.com/36999962/163548027-9608f277-4fab-42a1-b957-3d6515e72e00.PNG)

But we should see something like that (same joint but on the other leg):
![amo3](https://user-images.githubusercontent.com/36999962/163548088-0dba8ce7-f14a-4a1d-b5a7-f939f60566bf.PNG)

We checked also:

- AMO board and support orientation
- Electrical connections

We have to verify the correct functioning of the internal magnetic ring.

@maggia80 or @pattacini, do you have suggestions about that?

**Comment by davidetome** [comment]
@maggia80 I remember the same behavior when we mounted the magnetic rings reversed.
In this case, the ring was not moved from its place, but maybe it has been damaged in the crash?

cc @gsisinna @Uboldi80 @pattacini

**Comment by pattacini** [comment]
>In this case, the ring was not moved from its place, but maybe it has been damaged in the crash?

A visual inspection may also reveal some chipping on the ring.

**Comment by gsisinna** [comment]
We tried with @Uboldi80 an other AMO board and we got beautiful results!
![amo_OK](https://user-images.githubusercontent.com/36999962/163556774-77657456-c0d7-4fc8-b42a-e38bb2aa05a4.PNG)

We calibrated the AMO and this issue can be closed.

**Comment by maggia80** [comment]
> Today with @Uboldi80 and @davidetome we tried to calibrate the new AMO board. The first step of analogic calibration went well, but we found some problems with the Nonius Calibration (see [procedures](https://icub-tech-iit.github.io/procedures/tp-amo-calibration/)). We got this result: ![AMO](https://user-images.githubusercontent.com/36999962/163548027-9608f277-4fab-42a1-b957-3d6515e72e00.PNG)
> 

> 
> We checked also:
> 
> * AMO board and support orientation
> * Electrical connections
> 
> We have to verify the correct functioning of the internal magnetic ring.
> 
> @maggia80 or @pattacini, do you have suggestions about that?

I suspect that in this case the AMO was mounted 180 degrees wrt the correct position. In fact, it's the same effect you get when the magnetic target is mounted in the opposite direction.

**Comment by Uboldi80** [comment]
During the calibration we had 2 different problems.
1 the first is given by the fact that the AMO board can actually be mounted incorrectly (reversed by 180 degrees) as the mechanics allow it.
We became aware of this problem after a failed calibration attempt. This type of problem should be solved through a mechanical arrangement between the AMO board support and the frame seat.
2 The second problem was given by the fact that the new AMO board mounted on the robot was actually NOT WORKING.
We realized this as following 3 other unsuccessful calibrations we decided to move the reading board of the right leg to the left. Repeating the calibration solved the problem.
At this point we have replaced the board of the left leg with a new one (the last one) present in the spare parts set dedicated to the x-prize.
After this replacement and recalibration, the problem was resolved on Friday.
